### ISSUE TITLE
Creditmemo for an order that contains only one configurable product is not generated correctly

### ISSUE THREAD

**Issue by leonhelmus** [magento/magento2#36722]
### Preconditions and environment

- Magento version 2.4.4
- php 8.1

Functionality to check if its the last creditmemo does not work correctly. It's a similar issue like:
https://github.com/magento/magento2/issues/31143

### Steps to reproduce

1. Add configurable product to cart
2. Create order
3. Create invoice
4. When refunding add some custom code to check if it is the last creditmemo.

When calling the function isLast() to check if this is the last creditmemo:
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Sales/Model/Order/Creditmemo.php#L562

It also checks the dummy items while this should be skipped. This is the same issue as described in https://github.com/magento/magento2/issues/31143 same resolution also works for creditmemo.

### Expected result

if all products of configurable are refunded it the function isLast() should return a true.

### Actual result

if all products of configurable are refunded it the function isLast() should return a false, because of the dummy items.

### Additional information

Patch to resolve the issue. Same resolution can be found here:
https://github.com/magento/magento2/pull/31472

```
diff --git a/vendor/magento/module-sales/Model/Order/Creditmemo.php b/vendor/magento/module-sales/Model/Order/Creditmemo.php
--- a/vendor/magento/module-sales/Model/Order/Creditmemo.php	
+++ b/vendor/magento/module-sales/Model/Order/Creditmemo.php	(date 1673357465659)
@@ -563,6 +563,11 @@
     {
         $items = $this->getAllItems();
         foreach ($items as $item) {
+            $orderItem = $item->getOrderItem();
+            if ($orderItem->isDummy()) {
+                continue;
+            }
+
             if (!$item->isLast()) {
                 return false;
             }
```

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [X] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-November** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-November** [comment]
Verified the issue on Magento 2.4-develop instance and the issue is reproducible with the steps mentioned in the description. Hence confirming this issue.

**Comment by leonhelmus** [comment]
@engcom-November thanks for confirming the issue as soon as you did!

**Comment by hostep** [comment]
Closed without explanation why?

Probably fixed by [BUG#7632: Creditmemo for an order that contains only one configurable product is not generated correctly](https://github.com/magento/magento2/commit/cb7cd87f08de30e3f65e94a9183e8e5a2ef9c7dd)
